google-cloud-spanner-hibernate-dialect/src/main/java/com/google/cloud/spanner/hibernate/schema/SpannerTableStatements.java

```diff
@@ -47,20 +47,36 @@
 
   private final SpannerDialect spannerDialect;
 
+  private SpannerDatabaseInfo spannerDatabaseInfo;
+
   public SpannerTableStatements(SpannerDialect spannerDialect) {
     this.spannerDialect = spannerDialect;
   }
 
+  /**
+   * Initializes the {@link SpannerDatabaseInfo} which contains information about what tables and
+   * indices exist in the Spanner database.
+   *
+   * @param spannerDatabaseInfo the {@link SpannerDatabaseInfo} to load.
+   */
+  public void initializeSpannerDatabaseInfo(SpannerDatabaseInfo spannerDatabaseInfo) {
+    this.spannerDatabaseInfo = spannerDatabaseInfo;
```

Comment:
This doesn't really initialize anything. Also, is there risk of the `SpannerDatabaseInfo` getting stale throughout the life of this object?

Reply:
I prefer "initialize" just to be consistent how I used the term elsewhere - the idea just being that these are fields which could only be computed after the object was created and must be set now. Like you must "init" before you use the object.

No there is no risk because the initialize is called at the beginning of every create, drop, and update operation.